src/server/lib/audit-report.js

```diff
@@ -42,77 +50,120 @@ async function createObligationSheet (periodId) {
+          'Upload': upload.filename,
+          [COLUMN.EC_BUDGET]: 0,
+          [COLUMN.EC_TCO]: 0,
+          [COLUMN.EC_TCE]: 0,
+          [COLUMN.EC_CPO]: 0,
+          [COLUMN.EC_CPE]: 0,
+          [COLUMN.E50K_OBLIGATION]: 0,
+          [COLUMN.E50K_TEA]: 0,
+          [COLUMN.E_CPO]: 0,
+          [COLUMN.E_CPE]: 0
         }
-      }, emptyRow)
 
-      return row
-    }))
-  }))
+        const row = records
+          .filter(record => record.upload.id === upload.id)
+          .reduce((row, record) => {
+            switch (record.type) {
+              case 'ec1':
+              case 'ec2':
+              case 'ec3':
+              case 'ec4':
+              case 'ec5':
+              case 'ec7':
+                return {
+                  ...row,
+                  [COLUMN.EC_BUDGET]:
+                    row[COLUMN.EC_BUDGET] + record.content.Adopted_Budget__c,
+                  [COLUMN.EC_TCO]:
+                    row[COLUMN.EC_TCO] + record.content.Total_Obligations__c,
+                  [COLUMN.EC_TCE]:
+                    row[COLUMN.EC_TCE] + record.content.Total_Expenditures__c,
+                  [COLUMN.EC_CPO]:
+                    row[COLUMN.EC_CPO] +
+                    record.content.Current_Period_Obligations__c,
+                  [COLUMN.EC_CPE]:
+                    row[COLUMN.EC_CPE] +
+                    record.content.Current_Period_Expenditures__c
+                }
+              case 'awards50k':
+                return {
+                  ...row,
+                  [COLUMN.E50K_OBLIGATION]:
+                    row[COLUMN.E50K_OBLIGATION] + record.content.Award_Amount__c
+                }
+              case 'expenditures50k':
+                return {
+                  ...row,
+                  [COLUMN.E50K_TEA]:
+                    row[COLUMN.E50K_TEA] + record.content.Expenditure_Amount__c
+                }
+              case 'awards':
+                return {
+                  ...row,
+                  [COLUMN.E_CPO]:
+                    row[COLUMN.E_CPO] +
+                    record.content.Quarterly_Obligation_Amt_Aggregates__c,
+                  [COLUMN.E_CPE]:
+                    row[COLUMN.E_CPE] +
+                    record.content.Quarterly_Expenditure_Amt_Aggregates__c
+                }
+              default:
+                return row
+            }
+          }, emptyRow)
+
+        return row
+      }))
+    })
+  )
 
   return rows.flat()
 }
 
+async function createProjectSummaries (periodId) {
+  const records = await recordsForReportingPeriod(periodId)
+
+  const rows = []
+
+  records.forEach(record => {
+    switch (record.type) {
+      case 'ec1':
+      case 'ec2':
+      case 'ec3':
+      case 'ec4':
+      case 'ec5':
+      case 'ec7':
+        rows.push({
+          'Project ID': record.content.Project_Identification_Number__c,
+          'Upload': record.upload.filename // TODO: link to upload in arpa reporter
```

Comment:
i'm not sure how useful the filename will be, seems like maybe including at least an upload id column would be convenient, a link would be even better.

Reply:
Yeah, I'm imagining linking the filename to the upload in the tool.  👍 

I'll fix that as soon as we decide on a solution for links.